File: src/Tests/Behavioral/GlobalStructFieldPointers/features.go
package main

// Features is a package-global variable of an *anonymous* struct type, mirroring
// how the Go runtime declares internal/cpu.X86. The converter lifts this anonymous
// struct to a named C# type (e.g. "Featuresᴛ1") while visiting THIS file; the
// address of its fields is taken in main.go, which exercises cross-file resolution
// of that lifted type name. (Regression guard for the &cpu.X86.HasADX fix.)
var Features struct {
	HasFast bool
	HasWide bool
	Level   int
}

// option pairs a name with a pointer into a Features field, mirroring the
// internal/cpu.doinit options table built from "&X86.HasXXX" field addresses.
type option struct {
	name string
	flag *bool
}

File: src/Tests/Behavioral/GlobalStructFieldPointers/features.cs
namespace go;

partial class main_package {


[GoType("dyn")] partial struct Featuresᴛ1 {
    public bool HasFast;
    public bool HasWide;
    public nint Level;
}
public static ж<Featuresᴛ1> ᏑFeatures = new(default(Featuresᴛ1));
public static ref Featuresᴛ1 Features => ref ᏑFeatures.val;

[GoType] partial struct option {
    internal @string name;
    internal ж<bool> flag;
}

} // end main_package
